runconfig/hostconfig.go

```diff
@@ -217,6 +218,17 @@ func MergeConfigs(config *Config, hostConfig *HostConfig) *ContainerConfigWrappe
 	}
 }
 
+func ValidateHostConfig(hostConfig *HostConfig) error {
```

Comment:
This can go in `Daemon#verifyContainerSettings`:

https://github.com/docker/docker/blob/master/daemon/daemon_unix.go#L116